### PR TITLE
Improve explanation of escripts

### PR DESCRIPTION
**Pull request by bhargavrpatel**
### Context
As a reader, I would appreciate explicit mention of creating a Mix project; currently we mention mix casually. Additionally, I would have also liked a suggestion of where to place the `Module.CLI` module. This PR does that

### Changes
* Explain that you need to create a Mix project
* Explain the location of Module.CLI module
* Explain the generation of the executable

### CODE REVIEW

File: en/lessons/advanced/escripts.md [hunk 1]
```diff
@@ -11,11 +11,24 @@ To build executables in Elixir we will be using escript. Escript produces an exe
 
 ## Getting Started
 
-To create an executable with escript there are only a few things we need to do: implement a `main/1` function and update our Mixfile.
+For a given Mix project, to create an executable with escript there are only a few things we need to do: implement a `main/1` function and update our Mixfile.
+
+We'll begin by generating a project with a supervision tree:
+
+```shell
+$ mix new example_app
```
Comment (doomspork) [inline en/lessons/advanced/escripts.md]: This is incorrect.  To generate a project with a supervision tree we need:

```elixir
$ mix new example_app --sup
```

File: en/lessons/advanced/escripts.md [hunk 2]
```diff
@@ -11,11 +11,24 @@ To build executables in Elixir we will be using escript. Escript produces an exe
...
+$ cd example_app
+```
+
+We'll start by creating a module to serve as the entry point to our executable.  This is where we'll implement `main/1`. This module can be a separate file within the `lib` directory, or simply added to the default module like shown below.
```
Comment (doomspork) [inline en/lessons/advanced/escripts.md]: I'd like to keep what we had here already: "We'll start by creating a module to serve as the entry point to our executable.  This is where we'll implement `main/1`:"

By the time a reader has gotten this far in the lessons they should be familiar enough with things to know how to put this together.  We'll end up repeating too much if we review things in every lesson.

File: en/lessons/advanced/escripts.md [hunk 3]
```diff
@@ -11,11 +11,24 @@ To build executables in Elixir we will be using escript. Escript produces an exe
...
 
-We'll start by creating a module to serve as the entry point to our executable.  This is where we'll implement `main/1`:
 
 ```elixir
+# ./lib/example_app.ex
+defmodule ExampleApp do
```
Comment (doomspork) [inline en/lessons/advanced/escripts.md]: Please remove this.  We do not include this sort of placeholder in any other examples.

File: en/lessons/advanced/escripts.md [hunk 4]
```diff
@@ -26,6 +39,7 @@ end
 Next we need to update our Mixfile to include the `:escript` option for our project along with specifying our `:main_module`:
 
 ```elixir
+# ./mix.exs
```
Comment (doomspork) [inline en/lessons/advanced/escripts.md]: Please remove this, we do not use this format in any other lessons and we like to remain consistent.

**Comment by bhargavrpatel** [comment]
I could have sworn I've seen places where we define how to create a mix project in a non-mix specific document. However, I think what you said here makes sense. I am closing this out as there are not other changes.

**Pull request closed without merging**